### DISCUSSION TITLE
Should RecaptchaService be transient?

### DISCUSSION THREAD

**Discussion by sebastienros** [OrchardCMS/OrchardCore#14576] [General]
Since it's invoked by a filter, there is no reason to have it scoped (and added to a local dictionary).
But it could even be Singleton in order to cache the settings:
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.ReCaptcha/ReCaptchaLoginFilter.cs#L41

[the remark this answers is not in the document]
**Reply by jtkech** [reply]
A client is created by `ReCaptchaService` when its `VerifyAsync()` method is called. `ReCaptchaService` is injected by an `IAsyncResultFilter`, and also by an `ILoginFormEvent` and an `IPasswordRecoveryFormEvents`, so potentially the HttpClient could be used multiple times under the same request scope.

But currently only the `IAsyncResultFilter` call `VerifyAsync()` and even if the factory creates a new HttpClient more than once, the most important part is that the message handlers are pooled.

So, if you meant to not increase the scoped service registrations, yes it can be a transient or a singleton service.
